_posts/2017-03-28-bettracker-typescript.md

```diff
@@ -0,0 +1,116 @@
+---
+layout:     post
+title:      Modernizing our Application with TypeScript
+date:       2017-03-28 09:00
+summary:    How we used TypeScript to successfully refactor a large business critical codebase.
+category:   Software Engineering
+tags:       TypeScript, react, re-factoring
+author:     james_moorhouse
+---
+
+## Introduction
+
+Back in October 2016 my team were looking for ways we could improve our My Bets product which was written using ReactJS. We'd first made the product about 2 and a half years ago, much before my time, when ReactJS was still fairly new on the scene. As such the ReactJS itself has been through quite a few changes since then and a few of our design patterns were in need of a spring clean.
```

Comment:
Also, the library is just called "React" not "ReactJS" - that's the nomenclature [their docs](https://facebook.github.io/react/docs/thinking-in-react.html) uses, at least